tap_slack/__init__.py

```diff
@@ -2,28 +2,69 @@
 import json
 import singer
 from slack import WebClient
-from .streams import AVAILABLE_STREAMS
-from .catalog import generate_catalog
+
+from tap_slack.client import SlackClient
+from tap_slack.streams import AVAILABLE_STREAMS
+from tap_slack.catalog import generate_catalog
 
 LOGGER = singer.get_logger()
 
-def discover(webclient):
+
+def auto_join(client):
```

Comment:
If a user has specified to only sync specific channels (via the `channel` config), but also has `join_public_channels ` set to `true`, all public channels will be joined even though only a subset will be synced.

While this makes sense and the configs are fairly well named, it might still be a bit surprising. Should this only join the specified channels, if provided?

Alternatively changing the name of the config to `join_all_public_channels` would help, but even with that it might still be unexpected behaviour, as every Slack channel will suddenly have the bot join and they may not have the context!

Reply:
Good call. I'll change the logic to join public channels specified, or all if none are specified and document that.